crates/router/src/core/payments/helpers.rs

```diff
@@ -116,31 +116,19 @@ pub fn create_certificate(
         .change_context(errors::ApiClientError::CertificateDecodeFailed)
 }
 
-pub fn filter_mca_based_on_business_profile(
-    merchant_connector_accounts: Vec<domain::MerchantConnectorAccount>,
-    profile_id: Option<String>,
-) -> Vec<domain::MerchantConnectorAccount> {
-    if let Some(profile_id) = profile_id {
-        merchant_connector_accounts
-            .into_iter()
-            .filter(|mca| {
-                mca.profile_id.as_ref() == Some(&profile_id)
-                    && mca.connector_type == ConnectorType::PaymentProcessor
-            })
-            .collect::<Vec<_>>()
-    } else {
-        merchant_connector_accounts
-    }
-}
-
-pub fn filter_mca_based_on_connector_type(
-    merchant_connector_accounts: Vec<domain::MerchantConnectorAccount>,
-    connector_type: ConnectorType,
+pub fn filter_mca_based_on_profile_and_connector_type(
+    merchant_connector_accounts: &[domain::MerchantConnectorAccount],
+    profile_id: Option<&String>,
+    connector_type: Option<&ConnectorType>,
```

Comment:
why is this optinal? we are always passing this field wherever it is called right?